ghstack/cli.py

```diff
@@ -0,0 +1,235 @@
+import asyncio
+import contextlib
+from typing import Generator, List, Tuple
+
+import click
+
+import ghstack
+import ghstack.action
+import ghstack.checkout
+import ghstack.circleci_real
+import ghstack.config
+import ghstack.github_real
+import ghstack.land
+import ghstack.logs
+import ghstack.rage
+import ghstack.status
+import ghstack.submit
+import ghstack.unlink
+
+EXIT_STACK = contextlib.ExitStack()
+
+GhstackContext = Tuple[
+    ghstack.shell.Shell,
+    ghstack.config.Config,
+    ghstack.github_real.RealGitHubEndpoint,
+]
+
+
+@contextlib.contextmanager
+def cli_context(
+    request_circle_token: bool = False,
+) -> Generator[GhstackContext, None, None]:
+    with EXIT_STACK:
+        shell = ghstack.shell.Shell()
+        config = ghstack.config.read_config(request_circle_token=request_circle_token)
+        github = ghstack.github_real.RealGitHubEndpoint(
+            oauth_token=config.github_oauth,
+            proxy=config.proxy,
+            github_url=config.github_url,
+        )
+        yield shell, config, github
+
+
+@click.group(invoke_without_command=True)
+@click.pass_context
+@click.version_option(ghstack.__version__, "--version", "-V")
+@click.option("--debug", is_flag=True, help="Log debug information to stderr")
+# hidden arguments that we'll pass along to submit if no other command given
```

Comment:
Hmm, this is a little awkward. Should we make a helper function that lets us dedupe this between main and submit?

Reply:
yeah, I can build that out as a PR